### ISSUE TITLE
xgboost returns different classifiers when using incremental learning

### ISSUE THREAD

**Issue by klenhard** [dmlc/xgboost#3174]
Hey!

So, we're currently trying to train a classifier using the "multi:softprob" objective. As the classifier is trained on the pixel values of individual images, the classifier is trained one image at a time using the incremental training feature. The total dataset has a size of several thousand images.

However, we found that increasing the number of images that are trained on at a given time (i.e., incremental training with ~10 images at each training step) changes the structure of the output classifier considerably. Notably, the dump of the classifier trained incrementally on single images contains several >25k lines such as:

booster[846]:
0:leaf=0

compared to ~6k lines describing an actual decision tree. The decision tree of the other classifier does not include these lines (or orders of magnitude less).

The core xgboost calls for the training (which are looped over) are:

        dmat = xgb.DMatrix(data, label=labels, missing=0,
                           feature_names=data_header)

        xgb_model = xgb.train(
            params=params, dtrain=dmat, xgb_model=xgb_model)

Am I missing something / doing something wrong here? I'm using Python 2.7.12 and xgboost version 0.7.post3.

Thanks & Cheers!

**Comment by tqchen** [comment]
Questions are now moved to https://discuss.xgboost.ai/. In your case, xgboost uses a batch algorithm so it is indeed true that incrementally train with different dataset will results in different models
